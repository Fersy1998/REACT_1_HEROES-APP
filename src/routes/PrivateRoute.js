import React, { useContext } from 'react'
import { Navigate, useLocation } from 'react-router-dom';
import { AuthContext } from '../auth/authContext'



//Children son los componentes que envuelve PrivateRoute en AppRoutes
export const PrivateRoute = ({children}) => {

    //const location=useLocation();
    //desestructuramos y obtenemos el pathname y el search
    const {pathname, search}=useLocation();
    
    localStorage.setItem('last_path',pathname+search);
    const {user}=useContext(AuthContext);
    
    
  return user.logged ? children : <Navigate to='/login' />
            
}
