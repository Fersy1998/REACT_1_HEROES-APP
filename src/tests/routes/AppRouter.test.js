import { AppRoutes } from "../../routes/AppRoutes";
import React from 'react';
import {mount} from 'enzyme';
import { AuthContext } from "../../auth/authContext";
describe('pruebas en AppRoutes.js', ()=>{
    test('should show the search screen if the user is authenticated', () => {
        const contextValue={
            user:{
                name:'Fesry',
                logged:true
            }
            
        }
        const wrapper= mount(<AuthContext.Provider value={contextValue}>
                                <AppRoutes />
                            </AuthContext.Provider>);
        //console.log(wrapper.html());
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.navbar').exists()).toEqual(true);
        
    })
    test('should show the login component if the user isnÂ´t authenticated', () => {
        const contextValue={
            user:{
                logged:false
            }
            
        }
        const wrapper= mount(<AuthContext.Provider value={contextValue}>
                                <AppRoutes />
                            </AuthContext.Provider>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h5').text().trim()).toBe('Login Screen');
        
    })
    
    
    

})