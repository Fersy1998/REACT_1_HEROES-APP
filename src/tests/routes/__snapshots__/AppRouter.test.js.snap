// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en AppRoutes.js should show the login component if the user isn´t authenticated 1`] = `
<div
  className="container mt-5"
>
  <h5>
    Login Screen
  </h5>
     
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;

exports[`pruebas en AppRoutes.js should show the search screen if the user is authenticated 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end container"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link"
        >
          Fesry
        </span>
        <button
          className="nav-item nav-link btn btn-outline-warning"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <div
        className="row "
      >
        <h1>
          Hero List - 
          Marvel Comics
        </h1>
        <hr />
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-spider"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Spider Man"
              className="card-img-top"
              src="/assets/marvel-spider.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Spider Man
              </h5>
              <p
                className="card-text"
              >
                Amazing Fantasy #15
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Peter Parker
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-spider"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-captain"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Captain America"
              className="card-img-top"
              src="/assets/marvel-captain.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Captain America
              </h5>
              <p
                className="card-text"
              >
                Captain America Comics #1
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Steve Rogers
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-captain"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-iron"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Iron Man"
              className="card-img-top"
              src="/assets/marvel-iron.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Iron Man
              </h5>
              <p
                className="card-text"
              >
                Tales of Suspense #39
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Tony Stark
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-iron"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-thor"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Thor"
              className="card-img-top"
              src="/assets/marvel-thor.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Thor
              </h5>
              <p
                className="card-text"
              >
                Journey into Myster #83
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Thor Odinson
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-thor"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-hulk"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Hulk"
              className="card-img-top"
              src="/assets/marvel-hulk.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Hulk
              </h5>
              <p
                className="card-text"
              >
                The Incredible Hulk #1
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Bruce Banner
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-hulk"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-wolverine"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Wolverine"
              className="card-img-top"
              src="/assets/marvel-wolverine.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Wolverine
              </h5>
              <p
                className="card-text"
              >
                The Incredible Hulk #180
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                James Howlett
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-wolverine"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-daredevil"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Daredevil"
              className="card-img-top"
              src="/assets/marvel-daredevil.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Daredevil
              </h5>
              <p
                className="card-text"
              >
                Daredevil #1
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Matthew Michael Murdock
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-daredevil"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-hawkeye"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Hawkeye"
              className="card-img-top"
              src="/assets/marvel-hawkeye.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Hawkeye
              </h5>
              <p
                className="card-text"
              >
                Tales of Suspense #57
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Clinton Francis Barton
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-hawkeye"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-cyclops"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Cyclops"
              className="card-img-top"
              src="/assets/marvel-cyclops.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Cyclops
              </h5>
              <p
                className="card-text"
              >
                X-Men #1
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Scott Summers
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-cyclops"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
        <div
          className="col col-3 animate__animated animate__backInLeft"
          key="marvel-silver"
        >
          <div
            className="card  "
            style={
              Object {
                "width": "auto",
              }
            }
          >
            <img
              alt="Silver Surfer"
              className="card-img-top"
              src="/assets/marvel-silver.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Silver Surfer
              </h5>
              <p
                className="card-text"
              >
                The Fantastic Four #48
              </p>
              <p
                className="card-text"
              >
                Marvel Comics
              </p>
              <p
                className="card-text"
              >
                Norrin Radd
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-silver"
                onClick={[Function]}
              >
                Ver más
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
